packages/datetime/src/dateRangeInput.tsx

```diff
@@ -616,30 +627,38 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
         let nextState: IDateRangeInputState = { shouldSelectAfterUpdate: false };
 
         if (inputString.length === 0) {
-            // this case will be relevant when we start showing the hovered
-            // range in the input fields. goal is to show an empty field for
-            // clarity until the mouse moves over a different date.
+            // this case will be relevant when we start showing the hovered range in the input
+            // fields. goal is to show an empty field for clarity until the mouse moves over a
+            // different date.
+            const baseState = { ...nextState, [keys.inputString]: "" };
             if (isValueControlled) {
-                nextState = { ...nextState, [keys.inputString]: "" };
+                nextState = baseState;
```

Comment:
`baseState` seems only used here (not used in spread below) so please inline it.